### PR TITLE
Remove child controls

### PR DESCRIPTION
**Pull request by sapsh-msft**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

**Comment by sapsh-msft** [comment]
**Control Result/Logs/Detail logs will be drifted for following controls.**

1. SQLDatabase
	a. Azure_SQLDatabase_DP_Enable_TDE
	b. Azure_SQLDatabase_DP_Review_Data_Masking_Policy
	
2. ServiceFabric
	a. Azure_ServiceFabric_AuthN_NSG_Enabled 
	b. Azure_ServiceFabric_Audit_Use_Diagnostics_Log 
	c. Azure_ServiceFabric_Availability_Replica_Stateful_Size_Set_Min_3 
	
3. ServiceBus
	a. Azure_ServiceBus_AuthZ_Dont_Use_Policies_At_SB_Namespace
	b. Azure_ServiceBus_AuthZ_Use_Minimum_Access_Policies
	
4. EventHub
	a. Azure_EventHub_AuthZ_Dont_Use_Policies_At_SB_Namespace
    b. Azure_EventHub_AuthZ_Use_Min_Permissions_Access_Policies

File: src/AzSK/Framework/Core/SVT/Services/SQLDatabase.ps1 [hunk 1]
```diff
@@ -442,15 +329,17 @@ class SQLDatabase: SVTBase
 
 	hidden [ControlResult[]]  CheckSqlServerDataMaskingPolicy([ControlResult] $controlResult)
 	{
-
-		[ControlResult[]] $resultControlResultList = @()
+		
+		[string[]] $enabledDB = @()
+		[string[]] $disabledDB = @()
+		[string[]] $erroreddDB = @()
```
Comment (sbyna-msft) [inline src/AzSK/Framework/Core/SVT/Services/SQLDatabase.ps1]: check the typo

File: src/AzSK/Framework/Core/SVT/Services/SQLDatabase.ps1 [hunk 2]
```diff
@@ -470,42 +357,58 @@ class SQLDatabase: SVTBase
...
+				$controlResult.AddMessage([MessageData]::new("Database masking is enabled for following databases on SQL Server - ["+ $this.ResourceContext.ResourceName +"]",($enabledDB)));
+			}
+			if(($disabledDB | Measure-Object).Count -gt 0)
+			{
+				$controlResult.AddMessage([MessageData]::new("Database masking is disabled for following databases on SQL Server - ["+ $this.ResourceContext.ResourceName +"]",($disabledDB)));
+			}
+			if(($erroreddDB | Measure-Object).Count -gt 0)
+			{
+				$controlResult.AddMessage([MessageData]::new("Database masking is in error state for following databases on SQL Server - ["+ $this.ResourceContext.ResourceName +"]",($erroreddDB)));
+			}
```
Comment (sbyna-msft) [inline src/AzSK/Framework/Core/SVT/Services/SQLDatabase.ps1]: typo

File: src/AzSK/Framework/Core/SVT/Services/SQLDatabase.ps1 [hunk 3]
```diff
@@ -470,42 +357,58 @@ class SQLDatabase: SVTBase
...
+			if(($unknowndDB | Measure-Object).Count -gt 0)
+			{
+				$controlResult.AddMessage([MessageData]::new("Database masking is in unknown state for following databases on SQL Server - ["+ $this.ResourceContext.ResourceName +"]",($unknowndDB)));
```
Comment (sbyna-msft) [inline src/AzSK/Framework/Core/SVT/Services/SQLDatabase.ps1]: typo

File: src/AzSK/Framework/Core/SVT/Services/ServiceBus.ps1 [hunk 4]
```diff
@@ -2,9 +2,12 @@
 Set-StrictMode -Version Latest 
 class ServiceBus: SVTBase
 {       
-	hidden [PSObject[]] $NameSpacePolicies;
-	hidden [PSObject[]] $Queues;
-	hidden [PSObject[]] $Topics;
+	hidden [PSObject[]] $NameSpacePolicies = @() ;
```
Comment (sbyna-msft) [inline src/AzSK/Framework/Core/SVT/Services/ServiceBus.ps1]: check casing

File: src/AzSK/Framework/Core/SVT/Services/ServiceBus.ps1 [hunk 5]
```diff
@@ -15,41 +18,113 @@ class ServiceBus: SVTBase
...
+					# This block is intentionally left blank to handle exception while fetching topic access policies
+				}
+				
+			}
+		}
+
+		#endregion
+
+		$this.SBAccessPolicies = New-Object -TypeName PSObject 
+		$this.SBAccessPolicies | Add-Member -NotePropertyName NameSpacePolicies -NotePropertyValue $this.NameSpacePolicies
```
Comment (sbyna-msft) [inline src/AzSK/Framework/Core/SVT/Services/ServiceBus.ps1]: move the prop initialization to correct functions

File: src/AzSK/Framework/Core/SVT/Services/ServiceBus.ps1 [hunk 6]
```diff
@@ -15,41 +18,113 @@ class ServiceBus: SVTBase
...
+		$this.SBAccessPolicies | Add-Member -NotePropertyName Topics -NotePropertyValue ($this.TopicAccessPolicies)
+	}
+
+	hidden [ControlResult[]] CheckServiceBusRootPolicy([ControlResult] $controlResult)
+	{
+		$isControlFailed = $false
+		#region "NameSpace"
 		
-		$resultControlResultList += $childControlResult
+		$controlResult.SetStateData("Authorization rules for Service Bus namespace and child entities", $this.SBAccessPolicies);
```
Comment (sbyna-msft) [inline src/AzSK/Framework/Core/SVT/Services/ServiceBus.ps1]: check the attestation scenario @mprabhu11

File: src/AzSK/Framework/Core/SVT/Services/EventHub.ps1 [hunk 7]
```diff
@@ -52,48 +98,45 @@ class EventHub: SVTBase
...
-
 		#region "NameSpace"
-		[ControlResult] $childControlResult = [ControlResult]@{
-                            #ChildResourceName = $this.ResourceContext.ResourceName;
-                        };
-
-		$childControlResult.SetStateData("Authorization rules for Event Hub Namespace", $this.NameSpacePolicies);
-		$childControlResult.AddMessage([VerificationResult]::Verify, [MessageData]::new("Authorization rules for Namespace - ["+ $this.ResourceContext.ResourceName +"]. Validate that these rules are defined at correct entity level and with more limited permissions.", 
+		
+		$controlResult.SetStateData("Authorization rules for Eventhub namespace and child entities", $this.EHAccessPolicies);
```
Comment (sbyna-msft) [inline src/AzSK/Framework/Core/SVT/Services/EventHub.ps1]: check the attestation scenario with @mprabhu11

File: src/AzSK/Framework/Core/SVT/Services/ServiceBus.ps1 [hunk 8]
```diff
@@ -89,48 +158,44 @@ class ServiceBus: SVTBase
...
 			{
-				[ControlResult] $childControlResult = [ControlResult]@{
-						ChildResourceName = $topic.Name;
-					};
-
-				$topicPolicies = Get-AzureRmServiceBusAuthorizationRule -ResourceGroup $this.ResourceContext.ResourceGroupName `
-										-NamespaceName $this.ResourceContext.ResourceName -Topic $topic.Name
-
-				if(($topicPolicies|Measure-Object).count -gt 0)
+				if(($this.TopicAccessPolicies[$topic] |Measure-Object).count -gt 0)
```
Comment (sbyna-msft) [inline src/AzSK/Framework/Core/SVT/Services/ServiceBus.ps1]: check for keys

File: src/AzSK/Framework/Core/SVT/Services/ServiceFabric.ps1 [hunk 9]
```diff
@@ -145,93 +147,110 @@ class ServiceFabric : SVTBase
...
+			$controlResult.SetStateData("Diagnostics is disabled on Vmss", $diagnosticsDisabledScaleSet);
+		}
+		return $controlResult        
 	}
 
 	hidden [ControlResult[]] CheckStatefulServiceReplicaSetSize([ControlResult] $controlResult)
 	{
-		[ControlResult[]] $controlResultList = @() 
+		$isPassed = $true;
+		$complianteServices = @{};
```
Comment (sbyna-msft) [inline src/AzSK/Framework/Core/SVT/Services/ServiceFabric.ps1]: typos

File: src/AzSK/Framework/Core/SVT/Services/ServiceBus.ps1 (hunk 5 above)
Comment (sapsh-msft) [reply src/AzSK/Framework/Core/SVT/Services/ServiceBus.ps1]: Corrected.

File: src/AzSK/Framework/Core/SVT/Services/SQLDatabase.ps1 (hunk 2 above)
Comment (sapsh-msft) [reply src/AzSK/Framework/Core/SVT/Services/SQLDatabase.ps1]: Fixed.

File: src/AzSK/Framework/Core/SVT/Services/EventHub.ps1 [hunk 10]
```diff
@@ -4,46 +4,68 @@ class EventHub: SVTBase
...
+	hidden [void] GetEHAccessPolicies()
 	{
-		[ControlResult[]] $resultControlResultList = @()
 
-		#region "NameSpace"
-		[ControlResult] $childControlResult = [ControlResult]@{
-                            #ChildResourceName = $this.ResourceContext.ResourceName;
-                        };
+		#region "Event Hub"
+
```
[the remark this answers is not in the document]
Comment (sapsh-msft) [reply src/AzSK/Framework/Core/SVT/Services/EventHub.ps1]: Added try-catch block,

File: src/AzSK/Framework/Core/SVT/Services/EventHub.ps1 [hunk 11]
```diff
@@ -4,46 +4,92 @@ class EventHub: SVTBase
...
+				}
+			}
+		}
+        
+		#endregion
 
-		$childControlResult.AddMessage([VerificationResult]::Verify, [MessageData]::new("Following are the authorization rules for Namespace - ["+ $this.ResourceContext.ResourceName +"]. Validate that these rules must not be used at Event Hub level to send and receive messages.", 
-				$this.NameSpacePolicies));   
+		$this.EHAccessPolicies = New-Object -TypeName PSObject 
+		$this.EHAccessPolicies | Add-Member -NotePropertyName NameSpacePolicies -NotePropertyValue $this.NameSpacePolicies
```
[the remark this answers is not in the document]
Comment (sapsh-msft) [reply src/AzSK/Framework/Core/SVT/Services/EventHub.ps1]: Fixed.

File: src/AzSK/Framework/Core/SVT/Services/EventHub.ps1 [hunk 12]
```diff
@@ -102,32 +145,22 @@ class EventHub: SVTBase
...
-				if(($eventHubPolicies|Measure-Object).count -gt 0)
-				{
-					$childControlResult.SetStateData("Authorization rules for Event Hub:" + $eventHub.Name , $eventHubPolicies);
-					$childControlResult.AddMessage([VerificationResult]::Verify, [MessageData]::new("Authorization rules for Event Hub - ["+ $eventHub.Name +"]. Validate that these rules are defined at correct entity level and with more limited permissions.", $eventHubPolicies));
-				}
-				else
+				if(($this.EHChildAccessPolicies[$eventHub] |Measure-Object).count -gt 0)
 				{
-					$childControlResult.AddMessage([VerificationResult]::Passed, [MessageData]::new("No Authorization rules defined for Event Hub - ["+ $eventHub.Name +"]."));
+					$controlResult.AddMessage([MessageData]::new("Authorization rules for Event Hub - ["+ $eventHub.Name +"]. Validate that these rules are defined at correct entity level and with more limited permissions.", $this.EHChildAccessPolicies[$eventHub]));
```
[the remark this answers is not in the document]
Comment (sapsh-msft) [reply src/AzSK/Framework/Core/SVT/Services/EventHub.ps1]: Removed.

File: src/AzSK/Framework/Core/SVT/Services/SQLDatabase.ps1 [hunk 13]
```diff
@@ -116,108 +116,74 @@ class SQLDatabase: SVTBase
...
+			{
+				$controlResult.AddMessage([MessageData]::new("TDE is in error state for following databases on SQL Server - ["+ $this.ResourceContext.ResourceName +"]",($erroreddDB)));
+			}
+
 			if($atleastOneFailed) {
 				$controlResult.VerificationResult = [VerificationResult]::Failed;
+				
+				$DatabaseTDEFailed = New-Object -TypeName PSObject 
+				$DatabaseTDEFailed | Add-Member -NotePropertyName DisabledDB -NotePropertyValue $disabledDB
+				$DatabaseTDEFailed | Add-Member -NotePropertyName ErroredDB -NotePropertyValue $erroreddDB
```
[the remark this answers is not in the document]
Comment (sapsh-msft) [reply src/AzSK/Framework/Core/SVT/Services/SQLDatabase.ps1]: Fixed.

File: src/AzSK/Framework/Core/SVT/Services/SQLDatabase.ps1 (hunk 1 above)
Comment (sapsh-msft) [reply src/AzSK/Framework/Core/SVT/Services/SQLDatabase.ps1]: Fixed.

File: src/AzSK/Framework/Core/SVT/Services/SQLDatabase.ps1 (hunk 3 above)
Comment (sapsh-msft) [reply src/AzSK/Framework/Core/SVT/Services/SQLDatabase.ps1]: Fixed.

File: src/AzSK/Framework/Core/SVT/Services/ServiceBus.ps1 (hunk 4 above)
Comment (sapsh-msft) [reply src/AzSK/Framework/Core/SVT/Services/ServiceBus.ps1]: Fixed.

File: src/AzSK/Framework/Core/SVT/Services/ServiceBus.ps1 (hunk 8 above)
Comment (sapsh-msft) [reply src/AzSK/Framework/Core/SVT/Services/ServiceBus.ps1]: Added ContainsKey check.

File: src/AzSK/Framework/Core/SVT/Services/ServiceFabric.ps1 (hunk 9 above)
Comment (sapsh-msft) [reply src/AzSK/Framework/Core/SVT/Services/ServiceFabric.ps1]: Fixed.

**Pull request merged**